clients/java/src/main/java/io/camunda/client/impl/command/UpdateAuthorizationCommandImpl.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.client.impl.command;
+
+import io.camunda.client.api.CamundaFuture;
+import io.camunda.client.api.JsonMapper;
+import io.camunda.client.api.command.FinalCommandStep;
+import io.camunda.client.api.command.UpdateAuthorizationCommandStep1;
+import io.camunda.client.api.command.UpdateAuthorizationCommandStep1.UpdateAuthorizationCommandStep2;
+import io.camunda.client.api.command.UpdateAuthorizationCommandStep1.UpdateAuthorizationCommandStep3;
+import io.camunda.client.api.command.UpdateAuthorizationCommandStep1.UpdateAuthorizationCommandStep4;
+import io.camunda.client.api.command.UpdateAuthorizationCommandStep1.UpdateAuthorizationCommandStep5;
+import io.camunda.client.api.command.UpdateAuthorizationCommandStep1.UpdateAuthorizationCommandStep6;
+import io.camunda.client.api.response.UpdateAuthorizationResponse;
+import io.camunda.client.impl.http.HttpCamundaFuture;
+import io.camunda.client.impl.http.HttpClient;
+import io.camunda.client.protocol.rest.AuthorizationRequest;
+import io.camunda.client.protocol.rest.OwnerTypeEnum;
+import io.camunda.client.protocol.rest.PermissionTypeEnum;
+import io.camunda.client.protocol.rest.ResourceTypeEnum;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.hc.client5.http.config.RequestConfig;
+
+public class UpdateAuthorizationCommandImpl
+    implements UpdateAuthorizationCommandStep1,
+        UpdateAuthorizationCommandStep2,
+        UpdateAuthorizationCommandStep3,
+        UpdateAuthorizationCommandStep4,
+        UpdateAuthorizationCommandStep5,
+        UpdateAuthorizationCommandStep6 {
+
+  private final HttpClient httpClient;
+  private final JsonMapper jsonMapper;
+  private final RequestConfig.Builder httpRequestConfig;
+  private final AuthorizationRequest request;
+  private final long authorizationKey;
+
+  public UpdateAuthorizationCommandImpl(
+      final HttpClient httpClient, final JsonMapper jsonMapper, final long authorizationKey) {
+    this.httpClient = httpClient;
+    this.jsonMapper = jsonMapper;
+    httpRequestConfig = httpClient.newRequestConfig();
+    request = new AuthorizationRequest();
+    this.authorizationKey = authorizationKey;
+  }
+
+  @Override
+  public UpdateAuthorizationCommandStep2 ownerId(final String ownerId) {
+    ArgumentUtil.ensureNotNull("ownerId", ownerId);
+    ArgumentUtil.ensureNotEmpty("ownerId", ownerId);
```

Comment:
```suggestion
    ArgumentUtil.ensureNotNullNorEmpty("ownerId", ownerId);
```

🔧 We have a convenience method to put these both in a single call.